packages/samples/headless-ssr/src/app/page.tsx

```diff
@@ -1,8 +1,28 @@
 import {fetchInitialState} from '@/src/common/engine';
+import {
+  CoveoNextJsSearchParametersSerializer,
+  NextJSServerSideSearchParams,
+} from '@/src/common/search-parameters-serializer';
 import SearchPage from '@/src/components/search-page';
 
 // Entry point SSR function
-export default async function Search() {
-  const ssrState = await fetchInitialState();
+export default async function Search(url: {
+  searchParams: NextJSServerSideSearchParams;
+}) {
+  const {coveoSearchParameters} =
+    CoveoNextJsSearchParametersSerializer.fromServerSideUrlSearchParams(
+      url.searchParams
+    );
+  const ssrState = await fetchInitialState({
+    controllers: {
+      searchParameters: {
+        initialState: {
+          parameters: coveoSearchParameters,
+        },
+      },
+    },
+  });
   return <SearchPage ssrState={ssrState}></SearchPage>;
 }
+
+export const dynamic = 'force-dynamic';
```

Comment:
Next.js is supposed to automatically infer that the component should be rendered dynamically since it takes `searchParams` as an option, but for some reason it didn't, which resulted in prod builds not working.